docs/guides/chip_tool_guide.md

```diff
@@ -0,0 +1,808 @@
+commission a Matter device into the network and to communicate with it using
+Matter messages, which may encode Data Model actions, such as cluster commands.
+The tool also provides other utilities specific to Matter, such as parsing of
+the setup payload or performing discovery actions.
+
+<hr>
+
+-   [Source files](#source)
+-   [Building and running the CHIP Tool](#building)
+-   [Using the CHIP Tool for Matter device testing](#using)
+-   [Supported commands and options](#commands)
+
+<hr>
+
+<a name="source"></a>
+
+## Source files
+
+You can find source files of the CHIP Tool in the `examples/chip-tool`
+directory.
+
+> **Note:** The CHIP Tool currently only supports commissioning and remembering
+> one device at a time. The configuration state is cached in the
+> `/tmp/chip_tool_config.ini` file. Deleting this and other `.ini` files in the
+> `/tmp` directory can sometimes resolve issues related to stale configuration.
+
+<hr>
+
+<a name="building"></a>
+
+## Building and running the CHIP Tool
+
+Before you can use the CHIP Tool, you must compile it from source on Linux
+(amd64/aarch64) or macOS. If you want to build on Raspberry Pi, it must run
+Ubuntu 20.04.
+
+> **Note:** To ensure compatibility, always build the CHIP Tool and the Matter
+> device from the same revision of the `connectedhomeip` repository.
+
+### Building the CHIP Tool
+
+To build and run the CHIP Tool:
+
+1. Install all required packages for Matter and prepare the source code and the
+   build system. Read the [Building Matter](BUILDING.md) guide for instructions.
+2. Open a command prompt in the `connectedhomeip` directory.
+3. Run the following command:
+
+    ```
+    ./scripts/examples/gn_build_example.sh examples/chip-tool BUILD_PATH
+    ```
+
+    In this command, `BUILD_PATH` specifies where the target binaries are to be
+    placed.
+
+### Running the CHIP Tool
+
+To check if the CHIP Tool runs correctly, execute the following command from the
+`BUILD_PATH` directory:
+
+```
+$ chip-tool
+```
+
+As a result, the CHIP Tool prints all available commands. These are called
+_clusters_ in this context, but not all listed commands correspond to the
+_clusters_ in the Data Model (for example, pairing or discover commands). Each
+listed command can however become the root of the new more complex command by
+appending it with sub-commands. Examples of specific commands and their use
+cases are described in the [Supported commands and options](#commands) section.
+
+<hr>
+
+<a name="using"></a>
+
+## Using CHIP Tool for Matter device testing
+
+This section describes how to use CHIP Tool to test the Matter device. The
+following steps depend on the application clusters that you implemented on the
+device.
+
+This tutorial is using the
+[Matter lighting app example](https://github.com/project-chip/connectedhomeip/tree/master/examples/lighting-app)
+with the Bluetooth LE commissioning. You can use other Matter examples and still
```

Comment:
You also seem to cover "on-network" commissioning so BLE is not really an assumption, right?

Reply:
It's rather concerning the app itself. The following sentence aims to explain that it's a generic solution. I will try to clarify that.

Reply:
Ok, but the app can be run without BLE, too. Anyway, feel free to disregard that comment.